### PR TITLE
Lump dataset blocks

### PR DESCRIPTION
**Pull request by danielfdsilva**
Merges dataset blocks visually to allow for greater zooming out.

### CODE REVIEW

File: app/scripts/components/exploration/components/datasets/dataset-list-item.tsx [hunk 1]
```diff
@@ -367,51 +351,64 @@ function DatasetTrack(props: DatasetTrackProps) {
...
+          />
+          {wasLumped && (
+            <DatasetGLabel
+              fill='white'
+              x={Math.max(xScaled(blockStart), 2) + 2}
+              y={12}
+            >
+              G
+            </DatasetGLabel>
+          )}
```
Comment (nerik) [inline app/scripts/components/exploration/components/datasets/dataset-list-item.tsx:378]: Not a fan of the 'G' letter to signal a group. Is there a more visual way to express this? Like for instance:

- a "stacked" look
Screenshot 2023-09-14 at 09 34 39

- or an SVG pattern that reproduces what it already looks like when not lumped:
Screenshot 2023-09-14 at 09 36 00

File: app/scripts/components/exploration/components/datasets/block-utils.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+ * Lumps blocks of time together if they are too small to be seen on the chart.
+ *
+ * @param params.domain The dataset domain from which to calculate the blocks
+ * @param params.xScaled The xScaled function from the chart
+ * @param params.timeDensity The dataset time density
+ * @param params.minBlockSize The minimum size of a block
+ *
+ * @returns The blocks and whether or not any lumping happened.
+ */
+export function lumpBlocks({ domain, xScaled, timeDensity, minBlockSize = 4 }) {
```
Comment (nerik) [inline app/scripts/components/exploration/components/datasets/block-utils.ts:41]: Have we considered the "disjointed repetitive block" use case? (ie, every day of 2015 then every day of 2017). 
I think that's not allowed by the STAC API, but just checking.

File: app/scripts/components/exploration/components/datasets/dataset-list-item.tsx [hunk 3]
```diff
@@ -367,51 +351,64 @@ function DatasetTrack(props: DatasetTrackProps) {
...
-  const { xScaled, date, dataset, isVisible } = props;
+  const { xScaled, startDate, endDate, isVisible } = props;
 
   const theme = useTheme();
 
-  const [start, end] = getBlockBoundaries(date, dataset.data.timeDensity);
-  const s = xScaled(start);
-  const e = xScaled(end);
+  const s = xScaled(startDate);
+  const e = xScaled(endDate);
```
Comment (nerik) [inline app/scripts/components/exploration/components/datasets/dataset-list-item.tsx:398]: (nit=
```suggestion
  const xStart = xScaled(startDate);
  const xEnd = xScaled(endDate);
```

**Review by nerik — APPROVED** [review]
Works great 👍 
A few non-critical comments/questions, feel free to disregard.

File: app/scripts/components/exploration/components/datasets/dataset-list-item.tsx (hunk 1 above)
Comment (ricardoduplos) [reply app/scripts/components/exploration/components/datasets/dataset-list-item.tsx:378]: I like this "stacked" look suggestion a lot! CC/ @danielfdsilva

File: app/scripts/components/exploration/components/datasets/block-utils.ts (hunk 2 above)
Comment (danielfdsilva) [reply app/scripts/components/exploration/components/datasets/block-utils.ts:41]: Not sure I understood what you mean here

File: app/scripts/components/exploration/components/datasets/dataset-list-item.tsx (hunk 1 above)
Comment (danielfdsilva) [reply app/scripts/components/exploration/components/datasets/dataset-list-item.tsx]: This was a great suggestion!

**Pull request merged**